dbt/include/redshift/macros/materializations/materialized_view.sql

```diff
@@ -0,0 +1,106 @@
+    existing_relation,
+    backup_relation,
+    intermediate_relation
+) %}
+
+    -- apply a full refresh immediately if needed
+    {% if configuration_changes.requires_full_refresh %}
+
+        {{ get_replace_materialized_view_as_sql(relation, sql, existing_relation, backup_relation, intermediate_relation) }}
+
+    -- otherwise apply individual changes as needed
+    {% else %}
+
+        {%- set autorefresh = configuration_changes.autorefresh -%}
+        {%- if autorefresh -%}{{- log('Applying UPDATE AUTOREFRESH to: ' ~ relation) -}}{%- endif -%}
+
+        alter materialized view {{ relation }}
+            auto refresh {% if autorefresh.context %}yes{% else %}no{% endif %}
+
+    {%- endif -%}
+
+{% endmacro %}
+
+
+{% macro redshift__get_create_materialized_view_as_sql(relation, sql) %}
+
+    {%- set materialized_view = relation.from_runtime_config(config) -%}
+
+    create materialized view {{ materialized_view.path }}
+        backup {% if materialized_view.backup %}yes{% else %}no{% endif %}
+        diststyle {{ materialized_view.dist.diststyle }}
+        {% if materialized_view.dist.distkey %}distkey ({{ materialized_view.dist.distkey }}){% endif %}
+        {% if materialized_view.sort.sortkey %}sortkey ({{ ','.join(materialized_view.sort.sortkey) }}){% endif %}
+        auto refresh {% if materialized_view.auto_refresh %}yes{% else %}no{% endif %}
+    as (
+        {{ materialized_view.query }}
+    );
+
+{% endmacro %}
+
+
+{% macro redshift__get_replace_materialized_view_as_sql(relation, sql, existing_relation, backup_relation, intermediate_relation) %}
+    {{ redshift__get_drop_relation_sql(existing_relation) }};
+    {{ get_create_materialized_view_as_sql(relation, sql) }}
+{% endmacro %}
+
+
+{% macro redshift__get_materialized_view_configuration_changes(existing_relation, new_config) %}
+    {% set _existing_materialized_view = redshift__describe_materialized_view(existing_relation) %}
+    {% set _configuration_changes = existing_relation.materialized_view_config_changeset(_existing_materialized_view, new_config) %}
+    {% do return(_configuration_changes) %}
+{% endmacro %}
+
+
+{% macro redshift__refresh_materialized_view(relation) -%}
+    refresh materialized view {{ relation }}
+{% endmacro %}
+
+
+{% macro redshift__describe_materialized_view(relation) %}
+    {#-
+        These need to be separate queries because redshift will not let you run queries
+        against svv_table_info and pg_views in the same query. The same is true of svv_redshift_columns.
+    -#}
+
+    {%- set _materialized_view_sql -%}
+        select
+            tb.database,
+            tb.schema,
+            tb.table,
+            tb.diststyle,
+            tb.sortkey1,
+            mv.autorefresh
+        from svv_table_info tb
+        left join stv_mv_info mv
+            on mv.db_name = tb.database
+            and mv.schema = tb.schema
+            and mv.name = tb.table
+        where tb.table ilike '{{ relation.identifier }}'
```

Comment:
this needs to be `ilike`?

Reply:
I'm accounting for case-insensitivity, but maybe I shouldn't? It's probably worth thinking through that.

Reply:
For now I'm going to keep it since it aligns with `list_relations_without_caching`. But I do think it's worth rethinking it, just at a larger scope than MVs.